microprofile/WFLY-15405_microprofile_reactive_messaging_amqp.adoc

```diff
@@ -0,0 +1,128 @@
+= <INSERT TITLE HERE>
+:author:            Kabir Khan
+:email:             kkhan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+The scope of this RFE is to enhance the MicroProfile Reactive Messaging integration to include the AMQP connector in order to be able to send and receive data from AMQP message brokers.
+
+This is similar to the work already done to integrate with Kafka, and to be able to connect to an SSL secured AMQP broker we will leverage the same concepts to be able to use an SSLContext provided by the Elytron subsystem (see https://issues.redhat.com/browse/WFLY-14987[WFLY-14987]).
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-15405[WFCORE-15405]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-14987[WFLY-14987 - Securing the Kafka connector with an Elytron SSLContext]
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:{jstourac@redhat.com}[{Jan Stourac}]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+The changes will be made in the MicroProfile Reactive Messaging subsystem modules in the WildFLy codebase. SmallRye Reactive Messaging provides an AMQP Connector which will be included, and in turn relies on clients from Vert.x.
+
+The AMQPConnector uses the Vert.x clients, which in turn are based on Netty. This is designed on the concept of configuring an SSLContext by specifying paramters for protocols, truststore parameters etc. On inspection, there was no way to pass in a pre-defined SSLContext to this stack. After discussion with the relevant team leads, PRs have been opened to the following projects to allow for this:
+
+* https://github.com/smallrye/smallrye-reactive-messaging/pull/1741
+* https://github.com/vert-x3/vertx-amqp-client/pull/80
+* https://github.com/vert-x3/vertx-proton/pull/129
+* https://github.com/eclipse-vertx/vert.x/pull/4375
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+// Unchecking this since it was done for e.g. https://github.com/wildfly/wildfly-proposals/blob/main/microprofile/WFLY-14987_reactive_messaging-ssl-context-kafka-connector.adoc#issue and https://github.com/wildfly/wildfly-proposals/blob/main/microprofile/WFLY-14798-upgrade-reactive-messaging-2.0.adoc
```

Comment:
This feels wrong. I'm not asking you to change this analysis, but if a feature can work in **WildFly** in a domain and a user would reasonably expect it to work in a domain, then this box should be checked.

Hopefully this actually does work in a domain and this is just a paperwork thing.

Reply:
If the concern here is EAP oriented readers of this proposal, this works:

```
[X] Managed domain (note: downstream projects based on WildFly like EAP XP may not support managed domains)
```